pkg/frontend/querymiddleware/astmapper/astmapper.go

```diff
@@ -54,81 +54,71 @@ func NewMultiMapper(xs ...ASTMapper) *MultiMapper {
 	return m
 }
 
-// cloneNode is a helper function to clone a node.
-func cloneNode(node parser.Node) (parser.Node, error) {
-	return parser.ParseExpr(node.String())
+// cloneExpr is a helper function to clone an expr.
+func cloneExpr(expr parser.Expr) (parser.Expr, error) {
+	return parser.ParseExpr(expr.String())
 }
 
-func cloneAndMap(mapper ASTNodeMapper, node parser.Expr, stats *MapperStats) (parser.Node, error) {
-	cloned, err := cloneNode(node)
+func cloneAndMap(mapper ASTExprMapper, expr parser.Expr, stats *MapperStats) (parser.Expr, error) {
+	cloned, err := cloneExpr(expr)
 	if err != nil {
 		return nil, err
 	}
 	return mapper.Map(cloned, stats)
 }
 
-type NodeMapper interface {
-	// MapNode either maps a single AST node or returns the unaltered node.
+type ExprMapper interface {
+	// MapExpr either maps a single AST expr or returns the unaltered expr.
 	// It returns a finished bool to signal whether no further recursion is necessary.
-	MapNode(node parser.Node, stats *MapperStats) (mapped parser.Node, finished bool, err error)
+	MapExpr(expr parser.Expr, stats *MapperStats) (mapped parser.Expr, finished bool, err error)
 }
 
-// NewASTNodeMapper creates an ASTMapper from a NodeMapper
-func NewASTNodeMapper(mapper NodeMapper) ASTNodeMapper {
-	return ASTNodeMapper{mapper}
+// NewASTExprMapper creates an ASTMapper from a ExprMapper
+func NewASTExprMapper(mapper ExprMapper) ASTExprMapper {
+	return ASTExprMapper{mapper}
 }
 
-// ASTNodeMapper is an ASTMapper adapter which uses a NodeMapper internally.
-type ASTNodeMapper struct {
-	NodeMapper
+// ASTExprMapper is an ASTMapper adapter which uses a ExprMapper internally.
+type ASTExprMapper struct {
+	ExprMapper
 }
 
-// Map implements ASTMapper from a NodeMapper
-func (nm ASTNodeMapper) Map(node parser.Node, stats *MapperStats) (parser.Node, error) {
-	node, finished, err := nm.MapNode(node, stats)
+// Map implements ASTMapper from a ExprMapper
+func (nm ASTExprMapper) Map(expr parser.Expr, stats *MapperStats) (parser.Expr, error) {
+	expr, finished, err := nm.MapExpr(expr, stats)
 	if err != nil {
 		return nil, err
 	}
 
 	if finished {
-		return node, nil
+		return expr, nil
 	}
 
-	switch n := node.(type) {
+	switch n := expr.(type) {
 	case nil:
 		// nil handles cases where we check optional fields that are not set
 		return nil, nil
 
-	case parser.Expressions:
```

Comment:
How are `Expressions` handled in this new version? 🤔 
I assume `Map` will now be used for each singular `Expr`, but where is the traversal of expressions happening?

Reply:
Compiler told me we can't have `Expressions` :D
How do you get Expressions from an expression?

Reply:
I guess you are right 😅 If `Expressions` is `[]Expr`, it cannot be a single `Expr`. 
I'm just confused, because if an `Expr` is a wrapper around `Node`, how could a Node before be an `Expressions`?

Reply:
All expressions are nodes, but not all nodes are expressions.